src/pyfmi/fmi_util.pyx

```diff
@@ -205,17 +208,130 @@ cpdef prepare_data_info(np.ndarray[int, ndim=2] data_info, list sorted_vars, lis
         last_index = last_index + 1
         data_info[1,i] = last_index
 
-    data = np.append(model.time, np.concatenate(
-                                    (model.get_real(param_real),
-                                    model.get_integer(param_int).astype(float),
-                                    model.get_boolean(param_bool).astype(float),
-                                    np.array(diagnostics_param_values).astype(float)),
-                                    axis = 0
-                                )
-                    )
+    data = np.append(
+        model.time,
+        np.concatenate(
+            (model.get_real(param_real),
+             model.get_integer(param_int).astype(float),
+             model.get_boolean(param_bool).astype(float),
+             np.array(diagnostics_param_values).astype(float)
+            ),
+            axis = 0
+        )
+    )
 
     return data, varia_real, varia_int, varia_bool
 
+
+@cython.boundscheck(False)
+@cython.wraparound(False)
+cpdef prepare_data_info_fmi3(np.ndarray[int, ndim=2] data_info, list sorted_vars, list diagnostics_param_values, int nof_diag_vars, model):
+    cdef int index_fixed    = 1
+    cdef int index_variable = 1
+    cdef int nof_sorted_vars = len(sorted_vars)
+    cdef int nof_diag_params = len(diagnostics_param_values)
+    cdef int i, alias
+    cdef int last_data_matrix = -1, last_index = -1
+    cdef dict params = {
+        FMI3_Type.FLOAT64: [],
+        FMI3_Type.FLOAT32: [],
+        FMI3_Type.INT64: [],
+        FMI3_Type.INT32: [],
+        FMI3_Type.BOOL: [],
+        # TODO: Additional types
+    }
+    cdef dict variables = {
+        FMI3_Type.FLOAT64: [],
+        FMI3_Type.FLOAT32: [],
+        FMI3_Type.INT64: [],
+        FMI3_Type.INT32: [],
+        FMI3_Type.BOOL: [],
+        # TODO: Additional types
+    }
+    # "To set or inquire variables of type Enumeration, fmi3SetInt64 and fmi3GetInt64 must be used"
+    params[FMI3_Type.ENUM] = params[FMI3_Type.INT64]
```

Comment:
Not sure I get this; aren't these empty for now?
In the current format we'd still store any enum parameters in params[ENUM]?

Reply:
Ah, right.

How about explicitly copying the enum paramters & variables into corresponding int64 places after all variables/parameter have been sorted?

Reply:
I changed this again, I don't think the old approach was correct after thinking about it for a while. Initially that would cause some variables to be written to the result twice. Now instead with my latest commit:

1. Individual lists for int64 and enum
2. When concatenating the huge array with all the parameters I now simply added `model.get_int64(params[FMI3_Type.ENUM]).astype(float),` in the end.
3. I also temporarily added a `print(...)` in the function (not committed) `DumpDataFMI3.save_point` to verify only enums were written to result at the time and now I think it behaves as expected.